### PR TITLE
[1/5][main <- sdk] merge sdk branch into main and delete unwanted dep

### LINKED ISSUE #3: remove unused dep

### PR DESCRIPTION
**Pull request by VPhung24**
<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request includes changes to ESLint patterns, environment variables, and Next.js configuration. It adds environment variables and a webpack configuration, updates dependencies, and removes a test file. It also includes code changes related to logging, error handling, authentication, and smart contracts. The changes aim to improve the functionality and performance of the project.
> 
> ## What changed
> - Added environment variables and a webpack configuration
> - Updated dependencies and devDependencies
> - Deleted a test file
> - Added code for logging, error handling, authentication, and smart contracts
> 
> ## How to test
> 1. Clone the repository
> 2. Install the required dependencies
> 3. Run the project and verify that it functions as expected
> 4. Test the new features and functionalities introduced in this pull request
> 
> ## Why make this change
> This change was made to improve the project's functionality, performance, and maintainability. The addition of environment variables and a webpack configuration allows for better customization and optimization. The updates to dependencies ensure that the project is using the latest and most secure versions of external libraries. The removal of the test file indicates that it is no longer needed or relevant. The code changes related to logging, error handling, authentication, and smart contracts aim to enhance the project's overall functionality and user experience.
</details>

### CODE REVIEW

**Comment by VPhung24** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #6** <a href="https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/6?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #7** <a href="https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/7?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #8** <a href="https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/8?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/9?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/10?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/6?utm_source=stack-comment).

File: env.d.ts [hunk 1]
```diff
@@ -16,5 +16,6 @@ declare namespace NodeJS {
     BUGSNAG_API_KEY: string;
     NEXT_PUBLIC_DEBUG: string;
     NEXT_PUBLIC_APP_VERSION: string;
+    NEXT_PUBLIC_SHOW_TEST_VIEW: string;
```
Comment (ericlewis) [inline env.d.ts:19]: less action more declaration: NEXT_PUBLIC_TEST_VIEW_VISIBLE

Comment (VPhung24) [reply env.d.ts:19]: this is deleted in the next pull request

**Review by ericlewis — APPROVED** [review]

**Comment by VPhung24** [comment]
### Merge activity

* **Jan 19, 5:32 PM**: @VPhung24 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/6).
* **Jan 19, 5:32 PM**: @VPhung24 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/snowball-tools/iglootools-home/6).

**Pull request merged**